api/src/main/java/io/minio/PutObjectArgs.java

```diff
@@ -93,11 +93,8 @@ private void validatePartCount(int partCount, long objectSize, long partSize) {
       }
 
       if (partSize > 0) {
-        if (partSize > objectSize) {
-          partSize = objectSize;
-        }
-
-        long partCount = (long) Math.ceil((double) objectSize / partSize);
+        if (partSize > objectSize) partSize = objectSize;
+        long partCount = partSize > 0 ? (long) Math.ceil((double) objectSize / partSize) : 1;
```

Comment:
partsize is always greater than 0 here. Please look at line 95. So, not sure why this change is needed.

Reply:
line 96 may make it zero.